content/blog/posts/2021-03-11-getting-started-with-serverless.md

```diff
@@ -0,0 +1,174 @@
+
+```yaml
+functions:
+  graphql:
+    handler: handler.graphql
+    events:
+      - http:
+          path: graphql
+          method: post
+```
+
+For our needs, we'll want to also add a package for Graphql and I'm going to use [apollo-server-lambda](https://www.npmjs.com/package/apollo-server-lambda).
+
+```bash
+yarn add apollo-server-lambda
+```
+
+Now let's change our function to create a GraphQL server and use it to respond to query requests. For our demonstration, we'll just have it respond to a string for a simple query. 
+
+```js
+import { ApolloServer, gql } from 'apollo-server-lambda';
+
+const typeDefs = gql`
+  type Query {
+    hello: String
+  }
+`;
+
+const resolvers = {
+  Query: {
+    hello: () => 'Hello world!',
+  }
+};
+
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  introspection: true // this is nice in testing with GraphQL playground
+});
+
+export const graphql = server.createHandler();
+```
+
+## Try this out locally
+
+Now that you have a basic Lambda as your GraphQL server, you probably want to try it out before getting it out in the internet. Not that this is a requirement, most free tiers of providers make this a free experiment and so there's not risk to just pushing and testing. That said, a couple of packages also make it simple to test your function locally. 
+
+For our basic needs, we'll use the [serverless-offline](https://www.npmjs.com/package/serverless-offline) plugin to emulate AWS Lamdba and API Gateway services on a local server. In this regard we are choosing a vendor in a way, but for our purposes of chosing a path and pushing the code somewhere AWS is the one I'm personally most familiar with and that is the stated provider by default when generating a serverless project.
+
+Also, I like to write in module syntax, so adding [serverless-bundle](https://www.npmjs.com/package/serverless-bundle) to handle basic Webpack configuration for me gives me that flexibility. You can use it for adding Typescript, ESLint, Bable transpiling, and to generate source maps. For this use case though, I'll keep it simple with just our friend JavaScript.
+
+At this point, I need to add serverless locally as a dev dependency to my project as well, after the init used `npx` and does not add it at that time since the project didn't have a `package.json` until I added the first dependency.
+
+```bash
+yarn add serverless serverless-offline serverless-bundle -D
+```
+
+We then just add the config for the plugins to `serverless.yml` at the root:
+
+```yml
+plugins:
+  - serverless-bundle
+  - serverless-offline
+
+custom:
+  bundle:
+    packager: yarn
+  serverless-offline:
+    httpPort: 3000
+```
+
+After, we can test our function by running the offline start command. Note, I'm using `yarn` to prepend my command so that it looks first as local dependenicies for the executable.
```

Comment:
```suggestion
Next, we can test our function by running the offline start command. Note, I'm using `yarn` to prepend my command so that it looks first at local dependencies for the executable.
```